### PR TITLE
Calculating GCD and LCM by Recursion

### PR DESCRIPTION
**Pull request by Arun9739**
## Related Issue
- #1114 

## Proposed Changes
- I have added the code for finding the GCD and LCM by using recursion
- I have also updated the README.md

## Additional Info
- The answer given by the program was verified to be correct

## About the program 
- I have basically created two functions namely calculateGCD() and calculateLCM()
- calculateGCD() takes two integer inputs and returns their GCD
- It calculates the GCD recursively.
- Another function calculateLCM() calcultes the LCM by taking two inputs.
- The calculateLCM() function calls calculateGCD() so that it can directly calculate the LCM without any further iteration..
- I have updated the README.md file where I have explained in detail what is necessity of the recursion, how recusion works, what is the difference between the recursive approach and iterative approach. I have also mentioned the time complexity, space complexity, advantages and disadvantages.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| image  | image  |

### CODE REVIEW

**Comment by Arun9739** [comment]
@Mrjoy832 please review my pr. 
This is my **second** pr.
My first sucessfully merged pr is #1045

File: R-Recursion/README.md [hunk 1]
```diff
@@ -100,5 +100,40 @@ STOP
...
+# [Calculating GCD and LCM by Recursion](https://github.com/Arun9739/LearnCPP/blob/recursiveGCD/R-Recursion/Recursive_GCD_LCM.cpp)
 
+## Brief explainaion about GCD and LCM :
+The least common multiple (LCM) of two integers is the smallest positive integer that is a multiple of both. The greatest common divisor (GCD) of two integers is the largest positive integer dividing both. The product of the two numbers is the product of the LCM and the GCD.
 
+## What is Recursion : 
+The process in which a function calls itself directly or indirectly is called recursion and the corresponding function is called as recursive function. Using recursive algorithm, certain problems can be solved quite easily. Examples of such problems are Towers of Hanoi (TOH), Inorder/Preorder/Postorder Tree Traversals, DFS of Graph, etc.
+
+The Recursion can be demonstrted as base case and sub cases :
+![](https://files.realpython.com/media/Thinking-Recursively-in-Python_Watermarked.1825397c00ea.jpg)
```
Comment (Lakhankumawat) [inline R-Recursion/README.md:112]: rer

remove this image it is plagiarised from realpython

Rectify all the typos like shown in image from README , if exits .

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

**Comment by Arun9739** [comment]
@Lakhankumawat done the changes sir. Please review.

**Review by Lakhankumawat — APPROVED** [review]
LGTM ✨🎉

**Comment by Lakhankumawat** [comment]
I think you can earn Level-2 in your next PR show more efforts and keep contributing 🤞

**Pull request merged**